tests/aws/services/apigateway/test_apigateway_lambda_cfn.py

```diff
@@ -1,80 +1,304 @@
+                    },
+                }
+            ],
+        )
+
+        try:
+            aws_client.logs.create_log_group(logGroupName=f"/aws/lambda/{backend_function_name}")
+        except aws_client.logs.exceptions.ResourceAlreadyExistsException:
+            pass
+
+        try:
+            aws_client.logs.create_log_group(logGroupName=f"/aws/lambda/{oauth_function_name}")
+        except aws_client.logs.exceptions.ResourceAlreadyExistsException:
+            pass
+
+        aws_client.events.put_events(
+            Entries=[
+                {"Source": "source-123", "DetailType": "type-123", "Detail": json.dumps(detail)}
+            ]
+        )
+
+        def _check_backend_logs(retries=0):
+            events = get_lambda_log_events(
+                function_name=backend_function_name, logs_client=aws_client.logs
+            )
+            filtered = []
+            for e in events:
+                if isinstance(e, dict) and "body" in e:
+                    try:
+                        body = json.loads(e["body"])
+                        if body.get("marker") == unique_marker:
+                            filtered.append(e)
+                    except Exception as ex:
+                        print("Exception parsing event body:", ex)
+            if len(filtered) == 1:
+                return filtered[0]
+            raise AssertionError(
+                f"Expected exactly 1 invocation with marker {unique_marker}, found {len(filtered)}"
+            )
+
+        event_log = retry(_check_backend_logs, retries=30, sleep=5)
+        headers = {k.lower(): v for k, v in event_log.get("headers", {}).items()}
+        posted_body = json.loads(event_log.get("body", "{}"))
+        qs_params = event_log.get("queryStringParameters", {}) or {}
+
+        snapshot.match(
+            "final_invocation", {"headers": headers, "body": posted_body, "qs_params": qs_params}
+        )
+
+        assert posted_body["target_value"] == "value"
+        assert posted_body["marker"] == unique_marker
+        assert headers.get("target-header", "").lower() == "target_header_value"
+        assert qs_params.get("target_query") == "t_query"
+        assert qs_params.get("overwritten_query") == "original"
+
+        match auth_type:
+            case "BASIC":
+                user_pass = base64.b64encode(b"user:pass").decode("utf-8")
+                assert headers.get("authorization") == f"Basic {user_pass}"
+            case "API_KEY":
+                assert headers.get("api") == "apikey_secret"
+            case "OAUTH_CLIENT_CREDENTIALS":
+                assert headers.get("authorization") == "Bearer my-oauth-token"
+
+        aws_client.events.remove_targets(Rule=rule_name, Ids=[target_id])
+        aws_client.events.delete_rule(Name=rule_name, Force=True)
+        aws_client.events.delete_connection(Name=connection_name)
+        aws_client.events.delete_api_destination(Name=dest_name)
```

Comment:
nit: I think if anything fails in the test, this won't cleanup? Would it be better served as either a fixture or adding them to `cleanups`?